src/test/java/bio/terra/common/fixtures/DaoOperations.java

```diff
@@ -20,19 +24,25 @@ public class DaoOperations {
   private final DatasetDao datasetDao;
   private final ProfileDao profileDao;
   private final GoogleResourceDao resourceDao;
+  private final SnapshotDao snapshotDao;
+  private final SnapshotService snapshotService;
 
   public DaoOperations(
       JsonLoader jsonLoader,
       DatasetDao datasetDao,
       ProfileDao profileDao,
-      GoogleResourceDao resourceDao) {
+      GoogleResourceDao resourceDao,
+      SnapshotDao snapshotDao,
+      SnapshotService snapshotService) {
     this.jsonLoader = jsonLoader;
     this.datasetDao = datasetDao;
     this.profileDao = profileDao;
     this.resourceDao = resourceDao;
+    this.snapshotDao = snapshotDao;
+    this.snapshotService = snapshotService;
```

Comment:
Is it okay to add a service class to DaoOperations file? It was already being used in SnapshotDaoTest and I wanted to do the same thing for SnapshotBuilderSettingsDaoTest. So, I refactored it here because this is where datasets for the tests get created, so I thought it would make sense for snapshots to be made here too. But, just want to call this out in case we don't want to rely on a service class here.

Reply:
I think we should probably not include a service here. That feels like maybe we are moving outside of the unit test realm? It looks like the way it does datasets is by directly inserting via the DAO, could we do that with the snapshot too?

Reply:
So we do insert it directly in the DAO. It's the creation of the snapshot from the snapshotRequestModel that happens in the service, not insertion. I was following the example of how snapshots are created for SnapshotDaoTest. There is not a dao method to convert from SnapshotRequestModel to Snapshot, only a method in the service class. For a dataset, this process (converting from a datasetRequst to a dataset) is done in DatasetUtils.convertRequestWithGeneratedNames(datasetRequest). I could replicate the necessary functionality from the service method in a helper in the DaoOperations file or make a new SnapshotUtils, but I don't know if thats good option.

Reply:
I wonder if this class should be renamed `IntegrationOperations` as it's kind of an analog to `ConnectedOperations`? Since integration and connected tests work so differently it's a bit confusing to me that test fixtures for both are in the same package.

Reply:
Good idea! I will rename the class